### ISSUE TITLE
"Connect" button on "connected databases" tab in settings does nothing.

### ISSUE THREAD

**Issue by meta4** [keepassxreboot/keepassxc-browser#1219]
I have a KeePass database that I'm trying to reconnect to Google Chrome on an M1 Macbook Air.  I have been using Homebrew to manage both my KeyPassXC and Google Chrome software.  A week or two ago Homebrew, KeePassXC and Chrome upgraded to native Apple Silicon (AS).  Before that I ran them through Rosetta2.  Everything worked great.  I think I upgraded KeePassXC to AS first, but I upgraded Chrome relatively soon after that.  Between the two upgrades, I didn't notice whether or not keepassxc-browser could still connect to my database.  A few days ago, it wasn't connecting so I removed the connection on the connected database tab in settings.  I thought I would be able to reconnect the database, but that didn't work.  Now the connect button does nothing.  It's the connect button labeled "B" in step 5 of the instructions listed [here](https://keepassxc.org/docs/KeePassXC_GettingStarted.html#_configure_keepassxc_browser).

## Expected Behavior

I'm trying to follow the instructions [here](https://keepassxc.org/docs/KeePassXC_GettingStarted.html#_configure_keepassxc_browser).

## Current Behavior
When I get to step 6, pressing the connect button does nothing.  The button changes color when the mouse cursor hovers over it and highlights and changes color when clicked, but no dialog box appears.

## Possible Solution
I don't have any solution.

## Steps to Reproduce (for bugs)
I can't reproduce or fix it.  I'm assuming it has something to do with upgrading homebrew, KeePassXC, Chrome from X86_64 to AS.

## Debug info
I know how to debug python.  But I don't develop JavaScript much, and I don't have experience with the Chrome Developer Tools.
When I open the KeePassXC-Browser settings (chrome-extension://oboonakemofpalcgghocfoadofidjkkk/options/options.html) there are no errors in the console.  There are two warnings.

> DevTools failed to load SourceMap: Could not load content for chrome-extension://oboonakemofpalcgghocfoadofidjkkk/fonts/fork-awesome.min.css.map: HTTP error: status code 404, net::ERR_UNKNOWN_URL_SCHEME
> DevTools failed to load SourceMap: Could not load content for chrome-extension://oboonakemofpalcgghocfoadofidjkkk/bootstrap/bootstrap.min.js.map: HTTP error: status code 404, net::ERR_UNKNOWN_URL_SCHEME

When I click the connect button there is no change in either the errors tab or the warnings tab.

KeePassXC - 2.6.4
KeePassXC-Browser - 1.7.6
Operating system: MacIntel <== This is from the "Report Bugs" card on the about tab of settings.  I'm on an M1 MB air.
Browser: Chrome/Chromium 88.0.4324.150
OS Version: Big Sur 11.2.1

**Comment by meta4** [comment]
Things I've tried:

- Shutdown and restart Chrome
- Shutdown and restart KeePassXC
- Shutdown both Chrome and KeePassXC at the same time.
- Uninstall and re-install keepassxc-browser extension
- Uninstall and re-install KeePassXC
- Uninstall and re-install Google Chrome
- Reboot Mac.

**Comment by varjolintu** [comment]
Have you tried installing the Apple Silicon version of KeePassXC manually? You can download it from here: https://keepassxc.org/download/#mac

See if there's any difference.

**Comment by meta4** [comment]
Ok. Yes. That solved the problem.

Why?

Does homebrew install KeePassXC in the "wrong" place?

I'm going to keep the version installed from the DMG for now, but I would like to use homebrew if possible.

I'm willing to try more experiments if that will help.  I'm relatively new to mac and homebrew, but I'm trying to learn.

**Comment by varjolintu** [comment]
I'm not sure how the Homebrew version should differ from the official install. Maybe it's not up-to-date after all.

**Comment by meta4** [comment]
Looking into this a little more I did the following.

1. With the manual installer version of KeePassXC-2.6.4-2-arm64 installed, I verified that the keepassxc-browser plug in is functional.
2. Copied the contents of `/Applications/KeePassXC.app/Contents/` to a directory in my home folder named `manual-installer`. (There are no other folders or files in the `KeePassXC.app` directory.)
3. Uninstalled the manual installer version of KeePassXC by deleting the icon in Applications in the finder.
4. Installed KeePassXC using homebrew with `brew install --cask keepassxc`.  
    - I've been installing and uninstalling both the manual and homebrew versions.  One is "cached" in my downloads directory and the other is "cached" by homebrew.
    - During installation one of the messages said
```
Already downloaded: 
/Users/pete/Library/Caches/Homebrew/downloads/cdf9209ad5a6b54e6cb6fd52c7ac477fc8b1afa750883cd6322499da46df7ee8--KeePassXC-2.6.4-2-arm64.dmg
```
5. Ran `diff -rq /Applications/KeePassXC.app/Contents/manual-installer`.  There was no difference (no output).
6. Verified that the keepassxc-browser plugin is no longer functional.
7. Ran `diff Library/Caches/Homebrew/downloads/cdf9209ad5a6b54e6cb6fd52c7ac477fc8b1afa750883cd6322499da46df7ee8--KeePassXC-2.6.4-2-arm64.dmg Downloads/KeePassXC-2.6.4-2-arm64.dmg` to compare the two installers.  They are the same.

In Summary, it looks like both the manual installer and homebrew install the same files in /Applications/KeePassXC.app.  So, the problem is not in /Applications/KeePassXC.app.  What else does the KeePassXC installer configure?

**Comment by droidmonkey** [comment]
DMG's don't configure anything, they are just containers (like zip or tar files). If the DMG's are truly the same then there should be no reason why one works over the other.

**Comment by dbarr914** [comment]
I am also having the same issue. 

My web browser will not connect to the KeyPassXC database. I have followed these steps as well and haven't been able to resolve the issue.

I am using Brave browser.

I installed from the DMG on the website and the extension from the webstore.

KeePassXC - unknown
KeePassXC-Browser - 1.7.6
Operating system: MacIntel
Browser: Chrome/Chromium 88.0.4324.152

**Comment by droidmonkey** [comment]
For brave choose the chromium option in the browser settings.

**Comment by dbarr914** [comment]
> For brave choose the chromium option in the browser settings.

Yep, I selected all of them and nothing worked. I just resolve the issue by clicking the following buttons in the "Manage extension" settings in Brave.

![image](https://user-images.githubusercontent.com/19821759/107972461-e0ec9f00-6f81-11eb-8b69-a74aa4f53ec7.png)

Once I checked these boxes the Databases populated in the KeePassXC browser extension.

**Comment by droidmonkey** [comment]
OK so not KeePassXC, thanks for the ping back!

**Comment by soratobukuroneko** [comment]
I have also that issue on Gentoo with Firefox (build from source, not the binary package). Enabling all browsers in KeePassXC and allowing Run in private windows in Firefox doesn't help. I don't know how to debug that. At least, just opening the web console doesn't show anything.

**Comment by varjolintu** [comment]
@soratobukuroneko A JSON script file exists at `~/.mozilla/native-messaging-hosts`? And `keepassxc-proxy` binary is running?

**Comment by soratobukuroneko** [comment]
The file exists in that directory, but the proxy isn't running even after restarting keepassxc.

**Comment by varjolintu** [comment]
@soratobukuroneko Try running the proxy binary manually from shell. Does that work?

**Comment by soratobukuroneko** [comment]
No, that doesn't work.

**Comment by varjolintu** [comment]
Well, what does it say?

**Comment by soratobukuroneko** [comment]
There is no output at all

**Comment by varjolintu** [comment]
No output means it works normally :)

**Comment by OceanApart** [comment]
For Brave Browser choose the *Google Chrome* integration instead of **Brave**. And connect database in KeePassXC-Browser.

![image](https://user-images.githubusercontent.com/19257940/114643945-3090e200-9d09-11eb-82ef-0c6d2f439abb.png)

**Comment by varjolintu** [comment]
Anyone still having this problem?

**Comment by OceanApart** [comment]
> Anyone still having this problem?

Latest KeePassXC and browser extension，enable *Google Chrome* work for me, not *Brave*.

**Comment by varjolintu** [comment]
@OceanApart Thanks, but I'm referring to the original issue where teh Connect button doesn't do anything.

Seems this thread has been also divided to multiple issues. Closing it. Make a new issue if any problems persist.

**Comment by soratobukuroneko** [comment]
I was unable to use it before. I didn't change anything special, but doing Firefox/Icecat/Keepassxc upgrades as new release came.

I just tried again, and now it's working in both Firefox and Icecat
